client/a8c-for-agencies/sections/marketplace/checkout/checkout-v2.tsx

```diff
@@ -19,6 +21,11 @@ function CheckoutV2() {
 
 	const { selectedCartItems } = useShoppingCart();
 
+	// Fetch selected products by slug for site checkout
+	const { selectedProductsBySlug } = useProductsBySlug();
+
+	const siteId = getQueryArg( window.location.href, 'site_id' )?.toString();
```

Comment:
@yashwin can't we just check if there is selected products instead of checking for the site_id parameter? it seems the site_id is never used in the BD checkout flow.

Reply:
Yes, makes sense. Made the changes: https://github.com/Automattic/wp-calypso/pull/107768/commits/70e34f75652fd912424e95aa6be0f438502db875